benchmark/variable_benchmark.cpp

```diff
@@ -4,49 +4,150 @@
+                         benchmark::Counter::OneK::kIs1024);
+}
+static void Args_Variable_copy(benchmark::internal::Benchmark *b) {
+  b->Arg(10)->Arg(20);
 }
-BENCHMARK(BM_Variable_binary_with_Variable);
-BENCHMARK(BM_Variable_binary_with_VariableSlice);
+static void Args_Variable_copy_sparse(benchmark::internal::Benchmark *b) {
+  b->Range(1 << 5, 1 << 12);
+}
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate1D<float>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate2D<float>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate3D<float>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate4D<float>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate5D<float>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate6D<float>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate1D<double>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate2D<double>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate3D<double>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate4D<double>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate5D<double>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, Generate6D<double>)
+    ->Apply(Args_Variable_copy);
+BENCHMARK_TEMPLATE(BM_Variable_copy, GenerateSparse<float>)
+    ->Apply(Args_Variable_copy_sparse);
+BENCHMARK_TEMPLATE(BM_Variable_copy, GenerateSparse<double>)
+    ->Apply(Args_Variable_copy_sparse);
+
+/* static void BM_Variable_trivial_slice(benchmark::State &state) { */
+/*   auto var = makeVariable<double>({{Dim::Z, 10}, {Dim::Y, 20}, {Dim::X,
+ * 30}}); */
+
+/*   for (auto _ : state) { */
+/*     ConstVariableSlice view(var); */
+/*     Variable copy(view); */
+/*   } */
+/* } */
+/* BENCHMARK(BM_Variable_trivial_slice); */
+
+/* // The following two benchmarks "prove" that operator+ with a VariableSlice
+ * is */
+/* // not unintentionally converting the second argument to a temporary
+ * Variable. */
+/* static void BM_Variable_binary_with_Variable(benchmark::State &state) { */
+/*   auto var = makeVariable<double>({{Dim::Z, 10}, {Dim::Y, 20}, {Dim::X,
+ * 30}}); */
+/*   Variable a = var(Dim::Z, 0, 8); */
+
+/*   for (auto _ : state) { */
+/*     Variable b = var(Dim::Z, 1, 9); */
+/*     auto sum = a + b; */
+/*   } */
+/* } */
+/* static void BM_Variable_binary_with_VariableSlice(benchmark::State &state) {
+ */
+/*   auto b = makeVariable<double>({{Dim::Z, 10}, {Dim::Y, 20}, {Dim::X, 30}});
+ */
+/*   Variable a = b(Dim::Z, 0, 8); */
+
+/*   for (auto _ : state) { */
+/*     auto sum = a + b(Dim::Z, 1, 9); */
+/*   } */
+/* } */
+/* BENCHMARK(BM_Variable_binary_with_Variable); */
+/* BENCHMARK(BM_Variable_binary_with_VariableSlice); */
```

Comment:
A lot fo commented code here?

Reply:
Yes, that was the existing code that did not compile.

Reply:
Can you fix it? Probably all that is wrong is missing `{}` around the slicing arguments, which were not required in the past.

Reply:
:+1: I'll do that in a separate PR, along with adding the benchmarks to CI (just building, not execution).

Reply:
Why not fix it here? I think it is trivial.